plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewsMetastoreHelper.java

```diff
@@ -0,0 +1,242 @@
+        if (existing.isPresent()) {
+            if (!replace || !isPrestoView(existing.get())) {
+                throw new ViewAlreadyExistsException(schemaViewName);
+            }
+
+            metastore.replaceTable(schemaViewName.getSchemaName(), schemaViewName.getTableName(), table, principalPrivileges);
+            return;
+        }
+
+        try {
+            metastore.createTable(table, principalPrivileges);
+        }
+        catch (TableAlreadyExistsException e) {
+            throw new ViewAlreadyExistsException(e.getTableName());
+        }
+    }
+
+    public void dropView(ConnectorSession session, SchemaTableName schemaViewName)
+    {
+        if (getView(session, schemaViewName).isEmpty()) {
+            throw new ViewNotFoundException(schemaViewName);
+        }
+
+        try {
+            metastore.dropTable(schemaViewName.getSchemaName(), schemaViewName.getTableName(), true);
+        }
+        catch (TableNotFoundException e) {
+            throw new ViewNotFoundException(e.getTableName());
+        }
+    }
+
+    public List<SchemaTableName> listViews(ConnectorSession session, Optional<String> database)
+    {
+        return listDatabases(database).stream()
+                .flatMap(this::listViews)
+                .collect(toImmutableList());
+    }
+
+    public Map<SchemaTableName, ConnectorViewDefinition> getViews(ConnectorSession session, Optional<String> schemaName)
+    {
+        ImmutableMap.Builder<SchemaTableName, ConnectorViewDefinition> views = ImmutableMap.builder();
+        for (SchemaTableName name : listViews(session, schemaName)) {
+            try {
+                getView(session, name).ifPresent(view -> views.put(name, view));
+            }
+            catch (TrinoException e) {
+                if (e.getErrorCode().equals(TABLE_NOT_FOUND.toErrorCode())) {
+                    // Ignore view that was dropped during query execution (race condition)
+                }
+                else {
+                    throw e;
+                }
+            }
+        }
+        return views.buildOrThrow();
+    }
+
+    private List<String> listDatabases(Optional<String> database)
+    {
+        if (database.isPresent()) {
+            if (isHiveSystemSchema(database.get())) {
+                return ImmutableList.of();
+            }
+            return ImmutableList.of(database.get());
+        }
+        return metastore.getAllDatabases();
+    }
+
+    private Stream<SchemaTableName> listViews(String schema)
+    {
+        // Filter on PRESTO_VIEW_COMMENT to distinguish from materialized views
+        return metastore.getTablesWithParameter(schema, TABLE_COMMENT, PRESTO_VIEW_COMMENT).stream()
+                .map(table -> new SchemaTableName(schema, table));
+    }
+
+    public Optional<ConnectorViewDefinition> getView(ConnectorSession session, SchemaTableName viewName)
```

Comment:
`ConnectorSession session` parameter seems to be unused. 
Make sure that also the other methods calling this method drop it as well : `dropView` , `getViews`.